src/Domain/Handler/SubscriptionPurchasedEventHandler.ts

```diff
@@ -58,37 +61,42 @@ implements DomainEventHandlerInterface
     user: User,
     subscriptionName: SubscriptionName
   ): Promise<void> {
-    const roleName = this.subscriptionNameToRoleNameMap.get(subscriptionName)
+    const currentRoleName = (await user.roles)[0].name as RoleName
```

Comment:
hmm that made me think we have a design flaw. 

I think we need to change the event to not include `fromRole` since a user can have multiple roles. So the event would include just the role name that is added to the user.

Reply:
Sounds good, actually do we need to include roles in the event at all? They wouldn't be used by the clients to make the API request to /features, or would they?

Reply:
Hmm actually the clients won't need to use them - we'll get the user uuid from session when a user calls api to get features - and if we'll have the user then we'll have all it's roles and permissions.